lib/runnable.js

```diff
@@ -229,14 +208,14 @@ Runnable.prototype.clearTimeout = function() {
  */
 Runnable.prototype.resetTimeout = function() {
   var self = this;
-  var ms = this.timeout() || 1e9;
+  var ms = this.timeout();
 
-  if (!this._enableTimeouts) {
+  if (ms === 0) {
     return;
   }
   this.clearTimeout();
   this.timer = setTimeout(function() {
-    if (!self._enableTimeouts) {
```

Comment:
why don't you use `ms === 0` here as above?

Reply:
Flagged by [resetTimeout](https://github.com/mochajs/mocha/blob/master/test/unit/runnable.spec.js#L719) test. Sets 10, then reset, then sets 0.

Inside of the above `setTimeout`, `ms` is the initially cached 10 whereas `self.timeout()` is the correct 0.